src/platforms/windows/daemon/windowssplittunnel.cpp

```diff
@@ -0,0 +1,535 @@
+#include "windowssplittunnel.h"
+#include "../windowscommons.h"
+#include "../windowsservicemanager.h"
+#include "logger.h"
+#include <QNetworkInterface>
+#define PSAPI_VERSION 2
+#include <Windows.h>
+#include <psapi.h>
+#include <fwpmu.h>
+#include <initguid.h>
+#include <iphlpapi.h>
+#include <QCoreApplication>
+#include <QFileInfo>
+namespace {
+Logger logger(LOG_WINDOWS, "WindowsSplitTunnel");
+
+
+DEFINE_GUID(ST_FW_WINFW_BASELINE_SUBLAYER_KEY,
+    0xc78056ff, 0x2bc1, 0x4211, 0xaa, 0xdd, 0x7f, 0x35, 0x8d, 0xef, 0x20, 0x2d);
+
+DEFINE_GUID(ST_FW_PROVIDER_KEY,
+    0xe2c114ee, 0xf32a, 0x4264, 0xa6, 0xcb, 0x3f, 0xa7, 0x99, 0x63, 0x56, 0xd9);
+}
+
+WindowsSplitTunnel::WindowsSplitTunnel(QObject* parent): QObject(parent)
+{
+    if(!isInstalled()){
+         logger.log() << "Driver is not Installed, doing so";
+         auto handle = installDriver();
+         if(handle == INVALID_HANDLE_VALUE){
+            WindowsCommons::windowsLog("Failed to install Driver");
```

Comment:
should we do a quick return? Otherwise you end up calling `CloseServiceHandle(INVALID_HANDLE_VALUE);`

Reply:
Wondering if we should disable the feature if we are unable to install the driver.
What do you think? This is how linux works and I like it.

Reply:
What we really need is an API between the client and daemon that communicate which features are available. Such an API would clean up the Linux implementation of that quite a bit.